vortex-ipc/src/message_reader.rs

```diff
@@ -189,14 +189,11 @@ impl<R: VortexRead> MessageReader<R> {
             dtype,
             flatbuffer,
             |flatbuffer| {
-                root::<fb::Message>(flatbuffer)
-                    .map_err(VortexError::from)
-                    .map(|msg| msg.header_as_chunk().unwrap())
-                    .and_then(|chunk| {
-                        chunk
-                            .array()
-                            .ok_or_else(|| vortex_err!("Chunk missing Array"))
-                    })
+                unsafe { root_unchecked::<fb::Message>(flatbuffer) }
```

Comment:
In this class `load_next_message` already runs flatbuffer validation. At this point we know we have a valid `fb::Message`